daemons/fenced/pacemaker-fenced.c

```diff
@@ -1546,12 +1556,31 @@ main(int argc, char **argv)
 
     pcmk__serve_fenced_ipc(&ipcs, &ipc_callbacks);
 
+    pcmk__register_formats(NULL, formats);
+    rc = pcmk__output_new(&out, "log", NULL, argv);
+    if ((rc != pcmk_rc_ok) || (out == NULL)) {
+        crm_err("Can't log resource details due to internal error: %s\n",
+                pcmk_rc_str(rc));
+        crm_exit(CRM_EX_FATAL);
+    }
```

Comment:
If this fails, out (and fenced_data_set->priv) will be NULL, which is going to trip up somewhere.  It'll either hit a CRM_ASSERT or a NULL pointer dereference we're not guarding because we assume it'll always be fine.  We could protect the code immediately following this, but I bet there'll be something else.

On the other hand, I can't see why pcmk__output_new would fail unless something was seriously wrong.  We control that the formatters will be registered ahead of time, and we pass a static string that is the formatter we want to use.

Reply:
Makes sense